infer/component.go

```diff
@@ -110,3 +110,76 @@ func (rc *derivedComponentController[R, I, O]) Construct(
 			return res, err
 		})
 }
+
+// ** Implementations for creating an [InferredComponent] using existing Pulumi component programs. **
```

Comment:
New functions/types within `infer` to allow creating `InferredComponents` using existing Pulumi component programs. Expanding on the `infer` package also has the added benefit that provider authors can mix both custom and existing component resources, along with now component programs.